platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/reports/CrashReporter.kt

```
@@ -0,0 +1,235 @@
+            getConfigDetails(crashConfigFileContents) ?: let {
+                return Initialized.MalformedConfigFile("Malformed content at configuration file")
+            }
+
+        val sourceDirectory = File("${appContext.cacheDir.absolutePath}/${crashConfigDetails.rootPath}")
+        if (!sourceDirectory.exists() || !sourceDirectory.isDirectory) {
+            return Initialized.WithoutPriorCrash("Prior crash not found")
+        }
+
+        val destinationDirectory = File("$filesDir$DESTINATION_FILE_PATH").apply { if (!exists()) mkdirs() }
+
+        return runCatching {
+            findAndCopyCrashFile(
+                sourceDirectory,
+                destinationDirectory,
+                crashConfigDetails.extensionFileName,
+            )
+        }.getOrElse {
+            return Initialized.ProcessingFailure("Couldn't process crash files. ${it.message}")
+        }
+    }
+
+    @UiThread
+    private fun findAndCopyCrashFile(
+        sourceDirectory: File,
+        destinationDirectory: File,
+        fileExtension: String,
+    ): CrashReporterState {
+        val crashFile =
+            findCrashFile(sourceDirectory, fileExtension)
+                ?: let {
+                    return Initialized.WithoutPriorCrash("Crash file not found in the source directory")
+                }
+
+        val destinationFile = File(destinationDirectory, crashFile.toFilenameWithTimeStamp())
+        crashFile.copyTo(destinationFile, overwrite = true)
+
+        return if (destinationFile.exists()) {
+            Initialized.CrashReportSent("Crash file copied successfully")
+        } else {
+            Initialized.WithoutPriorCrash("No prior crashes found")
+        }
+    }
+
+    private fun getConfigDetails(crashConfigFileContent: String): ConfigDetails? =
+        runCatching {
+            val crashConfigDetails = crashConfigFileContent.split(",")
+            ConfigDetails(crashConfigDetails[0], crashConfigDetails[1])
+        }.getOrNull()
```

wouldn't you prefer to return an Init Error here as well

---

Ideally we could pass the string of the error back up

---

disregard this if we choose to remove the `message` in the status